src/test/java/io/jenkins/plugins/analysis/warnings/recorder/RecorderITest.java

```diff
@@ -0,0 +1,294 @@
+package io.jenkins.plugins.analysis.warnings.recorder;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.collections.api.list.ImmutableList;
+import org.eclipse.collections.impl.factory.Lists;
+import org.junit.Test;
+
+import com.gargoylesoftware.htmlunit.html.DomElement;
+import com.gargoylesoftware.htmlunit.html.HtmlForm;
+import com.gargoylesoftware.htmlunit.html.HtmlNumberInput;
+import com.gargoylesoftware.htmlunit.html.HtmlPage;
+
+import hudson.model.FreeStyleProject;
+import hudson.model.Result;
+import hudson.model.Project;
+
+import io.jenkins.plugins.analysis.core.model.AnalysisResult;
+import io.jenkins.plugins.analysis.core.testutil.IntegrationTestWithJenkinsPerSuite;
+import io.jenkins.plugins.analysis.warnings.Java;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Verify the info page of the Java parser.
+ *
+ * @author Tanja Roithmeier
+ */
+public class RecorderITest extends IntegrationTestWithJenkinsPerSuite {
+
+    private static final int HEALTHY = 1;
+    private static final int UNHEALTHY = 9;
+    private static final String ENABLED_HEALTH_INFO = "Enabling health report (Healthy=1, Unhealthy=9, Minimum Severity=LOW)";
+
+    /**
+     * Verifies the info page with one warning as input.
+     */
+    @Test
+    public void shouldCreateReportWithOneWarning() {
+
+        FreeStyleProject job = createFreeStyleProject();
+        copySingleFileToWorkspace(job, "javac_1Warning.txt", "javac.txt");
+
+        enableJavaWarnings(job, "javac.txt");
+
+        AnalysisResult result = scheduleBuildAndAssertStatus(job, Result.SUCCESS);
+        InfoPage info = new InfoPage(job);
+
+        assertThat(result.getTotalSize()).isEqualTo(1);
+        assertThat(info.getInfoMessages()).isEqualTo(result.getInfoMessages());
+        assertThat(info.getErrorMessages()).isEqualTo(result.getErrorMessages());
+    }
+
+    /**
+     * Verifies the info page with one warning and enabled health report as input.
+     */
+    @Test
+    public void shouldCreateReportWithOneWarningAndHealthReport90() {
+
+        FreeStyleProject job = createFreeStyleProject();
+        copySingleFileToWorkspace(job, "javac_1Warning.txt", "javac.txt");
+
+        enableJavaWarnings(job, "javac.txt");
```

Comment:
Das Erzeugen des Projekts in Methode auslagern und in den anderen Tests wiederverwenden (wenn es geht)